vllm/engine/output_processor/util.py

```diff
@@ -1,13 +1,11 @@
-from typing import List
+from typing import Iterable, List
 from typing import Sequence as GenericSequence
-from typing import Union
 
-from vllm.model_executor.layers.sampler import SamplerOutput
-from vllm.sequence import PoolerOutput, SequenceGroupOutput
+from vllm.sequence import SequenceGroupOutput
 
 
 def create_output_by_sequence_group(
-        outputs: GenericSequence[Union[SamplerOutput, PoolerOutput]],
+        outputs: GenericSequence[Iterable],
```

Comment:
This doesn't look right to me ... outputs shouldn't be a sequence of iterables?

Reply:
`SamplerOutput` and `PoolerOutput` are iterable, I believe. This was just a simpler way to express the requirements here since the code doesn't actually care about any details of the type otherwise

Reply:
Actually, my IDE (VSCode) flags `SamplerOutput` and `PoolerOutput` as not being iterable. We can solve this by adding a type annotation like this:

```
class SamplerOutput(...):

    if TYPE_CHECKING:
        def __iter__(self) -> Iterator[CompletionSequenceGroupOutput]:
            raise RuntimeError("Method should be automatically generated")
```

Reply:
interesting - i thought it made mypy happy, apologies! I'll update this.

Reply:
My IDE uses pyright for linting so it may actually pass on mypy.

Reply:
I changed how this works. Can you see if it works for you @DarkLight1337 ?

Reply:
Yes, this looks good, thanks!